### PR TITLE
Running for multiple pages

### PR DESCRIPTION
**Pull request by kristjands**
Added support for running multiple pages with different configuration

### CODE REVIEW

File: helpers/config.py [hunk 1]
```diff
@@ -1,10 +1,18 @@
-json_file = './JSON/reviewed.json'
+import json
+from helpers.page import Page
```
Comment (LukaAndrojna) [inline helpers/config.py:3]: Separate imports into blocks: standard lib, external libs, internal libs

File: helpers/config.py [hunk 2]
```diff
@@ -1,10 +1,18 @@
...
 
-DEFAULT_URL = 'https://www.nepremicnine.net'
+json_file = 'configuration.json'
 
 from_email = 'example@gmail.com'
-to_email = 'example2@gmail.com'
+to_email = ["example2@gmail.com"]
```
Comment (LukaAndrojna) [inline helpers/config.py]: All quotes in python files should be single quotes to keep consistency

File: helpers/config_defaults.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+href_parser = 'a'
+bs4_attrs = {"class":"oglas_container", "itemtype": "http://schema.org/ListItem"}
+
+info_attributes_bs4_block = 'div'
+info_attributes_bs4_attrs = ''
+info_attributes_bs4_class = 'more_info'
+
+summary_attributes_bs4_block = 'div'
+summary_attributes_bs4_attrs = ''
+summary_attributes_bs4_class = 'more_info'
```
Comment (LukaAndrojna) [inline helpers/config_defaults.py]: new line missing at the end of the file

File: helpers/configuration.json [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+        },
+        {
+            "key": "LjubljanaMesto",
+            "to_email": ["example5@gmail.com"],
+            "default_url": "https://www.nepremicnine.net",
+            "page_url": "https://www.nepremicnine.net/oglasi-prodaja/ljubljana-mesto/",
+            "mail_subject": "Nepremicnine update - Ljubljana mesto"
+        }
+    ]
+}
```
Comment (LukaAndrojna) [inline helpers/configuration.json]: new line missing at the end of the file

File: helpers/email_notification.py [hunk 5]
```diff
@@ -3,17 +3,17 @@
 from email.mime.text import MIMEText
 from email.mime.base import MIMEBase
 from email.mime.multipart import MIMEMultipart
+from helpers.page import Page
```
Comment (LukaAndrojna) [inline helpers/email_notification.py]: imports need to be separated again

File: helpers/listing.py [hunk 6]
```diff
@@ -0,0 +1,8 @@
+class Listing:
+    def __init__(self, page_url: str, summary: str, info: str):
+        self._page_url = page_url
+        self._summary = summary
+        self._info = info
+    
+    def report(self) -> str:
+        return f'{self._page_url}\n{self._info}\n{self._summary}\n'
```
Comment (LukaAndrojna) [inline helpers/listing.py]: add new line at the end of file

File: helpers/page.py [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+        self.info_attributes_bs4_block = page["info_attributes"]["bs4_block"] if "info_attributes" in page and "bs4_block" in page["info_attributes"] else defaults.info_attributes_bs4_block
+        self.info_attributes_bs4_attrs = page["info_attributes"]["bs4_attrs"] if "info_attributes" in page and "bs4_attrs" in page["info_attributes"] else defaults.info_attributes_bs4_attrs
+        self.info_attributes_bs4_class = page["info_attributes"]["bs4_class"] if "info_attributes" in page and "bs4_class" in page["info_attributes"] else defaults.info_attributes_bs4_class
+
+        self.summary_attributes_bs4_block = page["summary_attributes"]["bs4_block"] if "summary_attributes" in page and "bs4_block" in page["summary_attributes"] else defaults.summary_attributes_bs4_block
+        self.summary_attributes_bs4_attrs = page["summary_attributes"]["bs4_attrs"] if "summary_attributes" in page and "bs4_attrs" in page["summary_attributes"] else defaults.summary_attributes_bs4_attrs
+        self.summary_attributes_bs4_class = page["summary_attributes"]["bs4_class"] if "summary_attributes" in page and "bs4_class" in page["summary_attributes"] else defaults.summary_attributes_bs4_class
+
```
Comment (LukaAndrojna) [inline helpers/page.py]: new line

File: scraper.py [hunk 8]
```diff
@@ -3,40 +3,45 @@
...
-
-    #email_msg = listings.generate_email()
-    #if email_msg != '':
-    #    email_notification.send(email_msg)
-
-    with open(config.json_file, 'w') as fp:
-        json.dump(listings.get_reviewed_listings(), fp)
-
-
+    page_config: Page
```
Comment (LukaAndrojna) [inline scraper.py:14]: Never seen anything like this before

**Review by LukaAndrojna — COMMENTED** [review]
Looks good, besides the quotes and missing new lines 😁

File: scraper.py (hunk 8 above)
Comment (kristjands) [reply scraper.py:14]: I wanted to define that page_config is of type Page and not any other type. Only in this case VS Code advised me for properties in Page class.

Comment (LukaAndrojna) [reply scraper.py:14]: interesting, the more you know

**Pull request merged**